riscv-pac/src/result.rs

```diff
@@ -0,0 +1,48 @@
+use core::fmt;
+
+/// Convenience alias for the [Result](core::result::Result) type for the library.
+pub type Result<T> = core::result::Result<T, Error>;
+
+/// Represents error variants for the library.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Error {
+    /// Attempted out-of-bounds access.
+    IndexOutOfBounds {
+        index: usize,
+        min: usize,
+        max: usize,
+    },
+    /// Invalid field value.
+    InvalidValue {
+        field: &'static str,
+        value: usize,
+        bitmask: usize,
+    },
+    /// Invalid value of a register field that does not match any known variants.
+    InvalidVariant { field: &'static str, value: usize },
```

Comment:
Maybe we could differentiate errors regarding fields of a register and errors regarding the whole register/value. The idea is avoiding defining the `field` in the `riscv-peripheral` examples, as I think it is not necessary.

```suggestion
    /// Invalid value of a register field.
    InvalidFieldValue {
        field: &'static str,
        value: usize,
        bitmask: usize,
    },
    /// Invalid value of a register field that does not match any known variants.
    InvalidFieldVariant { field: &'static str, value: usize },
     /// Invalid value.
     InvalidValue {
        value: usize,
        bitmask: usize,
    },
    /// Invalid value that does not match any known variants.
    InvalidVariant(usize),
```

Reply:
> Maybe we could differentiate errors regarding fields of a register and errors regarding the whole register/value.

This makes sense to me. The value covers the entire structure, so this separation could be useful in other cases without subfields.